flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasModelVersionCol.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.StringParam;
+import org.apache.flink.ml.param.WithParams;
+
+/** Interface for the shared model version column param. */
+public interface HasModelVersionCol<T> extends WithParams<T> {
+    Param<String> MODEL_VERSION_COL =
+            new StringParam(
+                    "modelVersionCol",
+                    "The version of the model data that the input data is predicted with.",
+                    null);
```

Comment:
Would it be more usable to using "version" as the default value and always output this information? The reason is that I suppose adding an extra long column should not have an observable performance impact for nearline (not batch) prediction.

Reply:
I agree that it should not raise an observable performance impact here. I set the default value as `null` since it is consistent with the exiting prediction case. Is there any other benefits (except debugging) to output the model version?

Reply:
By `the exiting prediction case`, do you mean `HasPredictionCol`? Currently `HasPredictionCol` defaults to prediction. Can you clarify it here?

The benefit is to minimize the config that users need to tune for the common scenario targeted by algorithms that depends on `HasModelVersionCol`.

I suppose that for algorithms such as `OnlineStandardScaler`, their users will run the algorithm continuously and observe multiple models emitted by it. And it seems useful for users to know "which version of the model data that the input data is predicted with". Since there is observable benefit and negligible cost, it seems better to enable this column by default?

Reply:
Thanks for the explaination. I have changed the default value of `HasModelVersionCol` as `version`.